### PR TITLE
chore: fix pull queries in https://docs/concepts/queries.md

### PR DESCRIPTION
**Pull request by cprasad1**
### Description 
fix pull queries' description in https://docs/concepts/queries.md by deleting an entire section on `Pull query features and limitations` and just adding a link to https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/select-pull-query/ instead

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]

File: docs/concepts/queries.md [hunk 1]
```diff
@@ -68,32 +68,11 @@ Because materialized views are incrementally updated as new events arrive,
...
-- Pull queries are expressed using a strict subset of ANSI SQL.
-- You can execute pull queries on any derived table created by using the
-  CREATE TABLE AS SELECT statement.
-- For non-windowed aggregations, pull queries only support looking up events
-  by key.
-- WHERE clauses must have constraints that encompass all key column(s) for non-windowed tables.
 
-- In addition, windowed tables support bounds on `WINDOWSTART` and `WINDOWEND` using operators
-  `<=`, `<`, `=`, `>`, `>=`.
-- JOIN, PARTITION BY, GROUP BY and WINDOW clauses aren't supported.
```
Comment (AlanConfluent) [inline docs/concepts/queries.md:86]: Do we mention this in the other doc?

File: docs/concepts/queries.md [hunk 2]
```diff
@@ -68,32 +68,11 @@ Because materialized views are incrementally updated as new events arrive,
...
-- SELECT statements can contain column arithmetic and function calls.
```
Comment (AlanConfluent) [inline docs/concepts/queries.md:87]: Or this?  Might be good to call some of these general things out if we don't elsewhere.

**Review by AlanConfluent — APPROVED** [review]
Might want to add some of that info to the other doc, if it's not already there.

File: docs/concepts/queries.md (hunk 1 above)
Comment (cprasad1) [reply docs/concepts/queries.md:86]: We implicitly mention this by saying
```
Synopsis:
SELECT select_expr [, ...]
  FROM table
  [ WHERE where_condition ]
  [ AND window_bounds ];
  ```

Comment (cprasad1) [reply docs/concepts/queries.md:86]: I am explicitly adding some of this info in the other doc as well

Comment (cprasad1) [reply docs/concepts/queries.md:86]: done

**Pull request merged**